Add loading info and run-time image data pull, plus small fixes

imrehg opened the pull request:
## Pace

Loader info that shows that the site is working. We can choose other theme from https://github.hubspot.com/pace/docs/welcome/ as well, this seems to work okay.  I'm using the side bar percentage, as it seemed kinda nice.

Doesn't make the page load faster, but at least it shows that things are loading, while investigating how to speed things up.

## PlotLoader

Managed to squeeze out an async plot data loader. Files are now copied to `src/site/static/figures/`. When loading a new image, say `src/site/static/figures/this/that.json` use `<PlotLoader source={"this/that"} />` to get it loaded async.

Updated the links in all the source I can find, and I think I found a few plots that were called incorrectly before (e.g. finance plot for recruitment text), which should be fixed up, but need a look.

imrehg committed:
Add side loader info with pace.js

Can choose other theme from https://github.hubspot.com/pace/docs/welcome/
as well, this seems to work okay.

Doesn't make the page load faster, but at least it shows that
things are loading, while investigating how to speed thigns up.

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
mega commit to migrate to client-pulled plotly data

Plots are no longer in the main app json, but fragmented
and pulled separately.

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
Add trailing slash to table of contents items

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

tcbegley committed:
Use hooks for plotloader

imrehg committed:
simplify state logic in PlotLoader

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

tcbegley commented:
This is great! The page is still a bit unresponsive while loading, I wonder if we should hide content until the site has loaded? Or is that more annoying than helpful?

imrehg commented:
I think the unresponsiveness happens when someone quickly navigates, and Plotly is doing the actual loading. If there's a bit of a chance to preload/prefetch, then it's snappier a bit.

But definitely getting to the edge of my understanding what happens. Will look a bit more for any further wins.

imrehg committed:
Merge branch 'pace' into pace-hooks

imrehg committed:
PlotLoader rework for fewer variables and clarity

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
Merge pull request #101 from imrehg/pace-hooks

Use hooks for plotloader

imrehg committed:
fix react/html misunderstanding with element setting

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg commented:
Caught this once, so might worth keeping an eye out...

```
index.js:2177 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in PlotLoader (created by MDXCreateElement)
[..snip..]
```

imrehg committed:
fix unmounted status change issues (hopefully)

Based on https://www.debuggr.io/react-update-unmounted-component/

Referring to this error:

```
index.js:2177 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in PlotLoader (created by MDXCreateElement)
```

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
simplify PlotLoader's return for clarity

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
add id to Plotly plots for easier accessibility

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>

imrehg committed:
upgdate pace theme

Signed-off-by: Gergely Imreh <gergely.imreh@faculty.ai>
